pkg/identityserver/store/errors.go

```diff
@@ -63,6 +63,9 @@ var (
 	ErrUserSessionNotFound = errors.DefineNotFound(
 		"user_session_not_found", "user session with id `{session_id}` not found", "user_id",
 	)
+	ErrUnauthorizedAdminRemoval = errors.DefinePermissionDenied(
```

Comment:
I would not say this is unauthorized, as the rights of the user have been checked. It should be failed precondition - a (new) precondition of the delete RPC is that the user is not the last admin.

Compiled up:
`ErrLastAdmin = errors.DefineFailedPrecondition("last_admin", "user `{user_id}` is the last admin"`.

Reply:
I went with Unauthorized as there is no authorization for anyone to delete the last user but I agree that failed precondition is more readable and fits better the situation. Will change